File: php/testcode/cmdi_017.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_cmdi_multihop
function runCommand(string $cmd): string {
    $output = [];
    exec($cmd, $output); // vuln-code-snippet vuln-line php_cmdi_multihop
    return implode("\n", $output);
}

function cmdi017(BenchmarkRequest $req): BenchmarkResponse {
    $userCmd = "ls " . $req->param('dir');
    $result = runCommand($userCmd);
    return BenchmarkResponse::ok($result);
}
// vuln-code-snippet end php_cmdi_multihop

File: ruby/testcode/cmdi_017.rb
require_relative 'shared'

def build_scan_command(target)
  "nmap -sV " + target
end

# vuln-code-snippet start ruby_cmdi_multihop
def run_network_scan(req)
  target = req.param('target')
  cmd = build_scan_command(target)
  output = `#{cmd}` # vuln-code-snippet vuln-line ruby_cmdi_multihop
  BenchmarkResponse.ok(output)
end
# vuln-code-snippet end ruby_cmdi_multihop
